lib/postinstall.js

```diff
@@ -0,0 +1,115 @@
+var fs = require("fs");
+var path = require("path");
+var appRoot = "../../";
+
+/**
+ * Create dev tools gradle runtime entry
+ */
+function writeFabricServiceGradleHook() {
+    console.log("Install Onesignal-build-gradle hook.");
+    try {
+        if (!fs.existsSync(path.join(appRoot, "hooks"))) {
+            fs.mkdirSync(path.join(appRoot, "hooks"));
+        }
+        if (!fs.existsSync(path.join(appRoot, "hooks", "after-prepare"))) {
+            fs.mkdirSync(path.join(appRoot, "hooks", "after-prepare"));
+        }
+        var scriptContent =
+            `
+var path = require("path");
+var fs = require("fs");
+module.exports = function($logger, $projectData, hookArgs) {
+  var platform = hookArgs.platform.toLowerCase();
+
+  function updateAppGradleScript(file) {
+    var appBuildGradleContent = fs.readFileSync(file).toString();
+    if (!appBuildGradleContent.match(/.*onesignal.*/)) {
+      $logger.trace("Configuring Onesignal for Android");
+      var search = -1;
+
+      var repositories = appBuildGradleContent.match(/repositories \{([^}]+)\}/gm);
+      if (!repositories) {
+          return;
+      }
+      search = appBuildGradleContent.indexOf(repositories[0]) + repositories[0].length;
+      appBuildGradleContent = appBuildGradleContent.substr(0, search) + \`\n	      maven { url "https://plugins.gradle.org/m2/" }\n\` + appBuildGradleContent.substr(search);
+
+      search = appBuildGradleContent.indexOf("apply plugin: \\"com.android.application\\"");
+      if (search == -1) {
+          return;
+      }
+      appBuildGradleContent = appBuildGradleContent.substr(0, search + 39) + \`\napply plugin: "com.onesignal.androidsdk.onesignal-gradle-plugin"\n\` + appBuildGradleContent.substr(search + 39);
+
+      appBuildGradleContent = appBuildGradleContent + \`
+          configurations.all {
+              resolutionStrategy.eachDependency { DependencyResolveDetails details ->
```

Comment:
This `resolutionStrategy` will overwrite `onesignal-gradle-plugin`'s. Using `OneSignal-Gradle-Plugin` is preferred as it will pick the best version based on all dependencies in your project.  Using an exact version override of `details.useVersion '27.1.0'` like you have below is fragile as OneSignal and other plugins won't always be compatible with this exact version in the future.